samples/apps/research-assistant-react/backend/main.py

```diff
@@ -0,0 +1,690 @@
+            userId INTEGER NOT NULL,
+            profile TEXT,
+            timestamp DATETIME NOT NULL,
+            UNIQUE (userId)
+        )
+        """
+            )
+            conn.commit()
+            cursor.close()
+            conn.close()
+        finally:
+            lock.release()
+    except Exception as e:
+        print(f"Error creating table: {e}")
+
+
+create_message_table_if_not_exist()
+create_personalization_profile_table_if_not_exist()
+
+
+def query_db(query, args=(), one=False):
+    try:
+        try:
+            lock.acquire(True)
+            conn = sqlite3.connect("database.sqlite")
+            conn.row_factory = sqlite3.Row
+            cursor = conn.cursor()
+            cursor.execute(query, args)
+            result = cursor.fetchall()
+            cursor.close()
+            conn.close()
+            # Convert rows to dictionaries
+            json_result = [dict(row) for row in result]
+            return (json_result[0] if json_result else None) if one else json_result
+        finally:
+            lock.release()
+    except Exception as e:
+        print(f"Error querying table: {e}")
+
+
+def insert_db(query, args=()):
+    try:
+        try:
+            lock.acquire(True)
+            conn = sqlite3.connect("database.sqlite")
+            cursor = conn.cursor()
+            cursor.execute(query, args)
+            conn.commit()
+            cursor.close()
+            conn.close()
+        finally:
+            lock.release()
+    except Exception as e:
+        print(f"Error inserting into db table: {e}")
+
+
+def get_highest_msg_id_for_user_and_root(user_id, root_msg_id):
+    try:
+        try:
+            lock.acquire(True)
+            # Connect to the database (or create a new one if it doesn't exist)
+            conn = sqlite3.connect("database.sqlite")
+
+            # Create a cursor object to execute SQL commands
+            cursor = conn.cursor()
+
+            # Select the highest msgId from the messages table for the given userId and rootMsgId
+            cursor.execute(
+                "SELECT MAX(msgId) FROM messages WHERE userId = ? AND rootMsgId = ?",
+                (user_id, root_msg_id),
+            )
+
+            # Fetch the result
+            result = cursor.fetchone()
+
+            cursor.close()
+            # Close the connection
+            conn.close()
+
+            # Return the highest msgId if it exists, otherwise return -1
+            if result[0] is not None:
+                return result[0]
+            else:
+                return -1
+        finally:
+            lock.release()
+    except Exception as e:
+        print(f"Error getting highest msg_id from table table: {e}")
+
+
+def get_md5_hash(string):
+    import hashlib
+
+    return hashlib.md5(string.encode()).hexdigest()
+
+
+# Example usage:
+path_to_config_list = "./OAI_CONFIG_LIST"
+resulting_config = create_llm_config(path_to_config_list)
```

Comment:
If this is a global variable, capitalize and define on the top. Is it even being used?